### PR TITLE
INTYGFV-15630: Added filtering possibility for REKO status

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

File: web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java [hunk 1]
```diff
@@ -365,12 +361,82 @@ void shouldFilterDoctorIdsExcludedInFilter() {
...
+        }
+
+        @Test
+        void shouldNotFilterRekoStatusIncludedInFilter() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, Collections.emptyList(),
+                    List.of(RekoStatusType.REKO_1.toString(), RekoStatusType.REKO_2.toString(), RekoStatusType.REKO_3.toString()));
+
+            assertEquals(sickLeaves, actualSickLeaveList);
+            assertEquals(expectedSickLeave, actualSickLeaveList.get(0));
```
Comment (mww-sopra-steria) [inline web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java:426]: Redundant asserts? assertEquals(sickLeaves, actualSickLeaveList); already checks the return data.

File: web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java [hunk 2]
```diff
@@ -365,12 +361,82 @@ void shouldFilterDoctorIdsExcludedInFilter() {
...
+            assertEquals(anotherSickLeave, actualSickLeaveList.get(1));
+        }
+
+        @Test
+        void shouldFilterSickLeaveWithNoRekoStatusAsRekoStatus1() {
```
Comment (mww-sopra-steria) [inline web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java:431]: Is there a sick leave that doesn't have a reko-status in the test data?

File: web/src/main/java/se/inera/intyg/intygstjanst/web/service/dto/GetSickLeaveServiceRequest.java [hunk 3]
```diff
@@ -41,4 +41,5 @@ public class GetSickLeaveServiceRequest {
     String protectedPersonFilterId;
     LocalDate fromSickLeaveEndDate;
     LocalDate toSickLeaveEndDate;
+    List<String> rekoStatuses;
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/intygstjanst/web/service/dto/GetSickLeaveServiceRequest.java:44]: Suggest change of name to rekoStatusTypeIds. This change should be made in necessary places.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Have a suggestion of a name-change for clarity. Do you agree? 

Also, take a look at the test comment concerning the test data used.

File: web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java (hunk 2 above)
Comment (mhornfeldt) [reply web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java:431]: sickLeaveWithNoRekoStatus doesn't have a rekoStatus.

File: web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImpl.java [hunk 4]
```diff
@@ -44,16 +45,28 @@ public FilterSickLeavesImpl(DiagnosisChapterService diagnosisChapterService, Cal
...
             .filter(sickLeave -> filterOnSickLeaveLengthIntervals(sickLeave, sickLeaveLengthIntervals))
             .filter(sickLeave -> filterOnDiagnosisChapters(sickLeave, diagnosisChapters))
             .filter(sickLeave -> filterOnPatientAge(sickLeave, fromPatientAge, toPatientAge))
             .filter(sickLeave -> filterOnSickLeaveEndDate(sickLeave, fromSickLeaveEndDate, toSickLeaveEndDate))
             .filter(sickLeave -> filterOnDoctorIds(sickLeave, doctorsIds))
+            .filter(sickLeave -> filterOnRekoStatuses(sickLeave, rekoStatusTypeIds))
             .collect(Collectors.toList());
     }
 
+    private boolean filterOnRekoStatuses(SjukfallEnhet sickLeave, List<String> rekoStatuses) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImpl.java:59]: rekoStatuses -> rekoStatusTypeIds (for consistency)

File: web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java (hunk 2 above)
Comment (mww-sopra-steria) [reply web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java:429]: Yes, but the test should include a sick leave without a rekoStatus

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Two small things :-) Not sure if I'm misunderstanding the test...

File: web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java (hunk 2 above)
Comment (mhornfeldt) [reply web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java:429]: All tests have the same sickleaves so all tests have sickLeaveWithNoRekoStatus in the test data

**Review by mww-sopra-steria — APPROVED** [review]

**Pull request merged**